operator/src/main/java/org/keycloak/operator/controllers/KeycloakDeployment.java

```diff
@@ -152,6 +153,11 @@ public void validatePodTemplate(KeycloakStatusBuilder status) {
             overlayTemplate.getSpec().getContainers().get(0).getImage() != null) {
             status.addWarningMessage("The image of the keycloak container cannot be modified using podTemplate");
         }
+
+        if (overlayTemplate.getSpec() != null &&
+                CollectionUtil.isNotEmpty(overlayTemplate.getSpec().getImagePullSecrets())) {
+            status.addWarningMessage("The imagePullSecrets of the keycloak container cannot be modified using podTemplate");
+        }
```

Comment:
One last thing. :) Since we have now added this warning message, can you also please add a test for it? It should be   similar to this:
https://github.com/keycloak/keycloak/blob/006aea300d7892fd60cf756535f25d2c325b3d1c/operator/src/test/java/org/keycloak/operator/testsuite/integration/PodTemplateTest.java#L158-L182

Reply:
Done! Thanks.